src/lib.rs

```diff
@@ -669,6 +669,67 @@ macro_rules! quote_bind_next_or_break {
+//   which this implementation avoids because it isn't tail recursive.
+// - Compile times for a tt muncher are quadratic relative to the length of the
+//   input. This implementation is linear, so it will be faster (potentially
+//   much faster) for long inputs. However, the constant factors of this
+//   implementation are higher than that of a tt muncher, so it is somewhat
+//   slower than a tt muncher if there are many invocations with short inputs.
+//
+// An invocation like this:
+// ```
+// quote_each_token! { _s a b c d e f g h i j }
+// ```
+// expands to this:
+// ```
+// quote_tokens_with_context! (_s
+//   (@   @   @  @   @   @   a   b   c   d   e   f   g  h   i   j)
+//   (@   @   @  @   @   a   b   c   d   e   f   g   h  i   j   @)
+//   (@   @   @  @   a   b   c   d   e   f   g   h   i  j   @   @)
+//   (@   @   @ (a) (b) (c) (d) (e) (f) (g) (h) (i) (j) @   @   @)
+//   (@   @   a  b   c   d   e   f   g   h   i   j   @  @   @   @)
+//   (@   a   b  c   d   e   f   g   h   i   j   @   @  @   @   @)
+//   (a   b   c  d   e   f   g   h   i   j   @   @   @  @   @   @)
+// );
+// ```
+// which expands to this:
+// ```
+// quote_token_with_context! (_s @ @ @  @  @ @ a);
+// quote_token_with_context! (_s @ @ @  @  @ a b);
+// quote_token_with_context! (_s @ @ @  @  a b c);
+// quote_token_with_context! (_s @ @ @ (a) b c d);
+// quote_token_with_context! (_s @ @ a (b) c d e);
+// quote_token_with_context! (_s @ a b (c) d e f);
+// quote_token_with_context! (_s a b c (d) e f g);
+// quote_token_with_context! (_s b c d (e) f g h);
+// quote_token_with_context! (_s c d e (f) g h i);
+// quote_token_with_context! (_s d e f (g) h i j);
+// quote_token_with_context! (_s e f g (h) i j @);
+// quote_token_with_context! (_s f g h (i) j @ @);
+// quote_token_with_context! (_s g h i (j) @ @ @);
+// quote_token_with_context! (_s h i j  @  @ @ @);
+// quote_token_with_context! (_s i j @  @  @ @ @);
+// quote_token_with_context! (_s j @ @  @  @ @ @);
+// ``
+// Without using recursion, we get one invocation of
+// `quote_token_with_context!` for each tt, with three tts of context on either
+// side. This is enough for the longest possible interpolation form (a
+// repetition interpolation with a separator, of the form `# (var) , *`) to be
+// fully represented with the first or last tt in the middle.
+//
+// The middle tt (surrounded by parentheses) is the tt being processed.
+// - When it is a `#`, `quote_token_with_context!` can do an interpolation. The
+//   interpolation kind will depend on the three subsequent tts.
+// - When it is within a later part of an interpolation, it can be ignored,
+//   because the interpolation has already been done.
+// - When it is not part of an interpolation it can be treated normally.
+// - When it is a `@` it can be ignored.
+//
+// The parentheses around the middle tt aren't necessary, but they make the
+// macro definitions easier to read.
```

Comment:
This part isn't right. I don't know how we would correctly drop the first 3 and last 3 quote_token_with_context calls without bracketing the middle token to make it unable to collide with the padding token (`@`).

I'll fix this in a followup PR.

Reply:
Ok, thanks!